### ISSUE TITLE
Do you have plan to upload the trained model in HF?

### ISSUE THREAD

**Issue by p1k0pan** [NJUNLP/njuqe#3]
Hi, thank you for your wonderful work. I wonder whether you guys have plan to release the model into HF?

**Comment by hy5468** [comment]
We plan to proceed with the open-sourcing of the model after the ICLR deadline. If you need the model, we can send it to you in advance.

**Comment by p1k0pan** [comment]
> We plan to proceed with the open-sourcing of the model after the ICLR deadline. If you need the model, we can send it to you in advance.

Yes, that would be nice. 
But how to send a model with a large data?

**Comment by hy5468** [comment]
We can work this out together. Please feel free to email me at gx@smail.nju.edu.cn for further discussion or assistance.

**Comment by hy5468** [comment]
We are pleased to announce the release of our DCSQE model, available at [DCSQE](https://huggingface.co/collections/DreamW1ngs/dcsqe-68a6885d0b392a1fbd2ae897). 

- The "syn" variant indicates a model pre-trained exclusively on pseudo data;
- The "sent" variant refers to a model that has undergone pre-training on pseudo data and additional fine-tuning on real QE data, with a focus on sentence-level tasks;
- The "word" variant denotes a model that has been pre-trained on pseudo data and further fine-tuned on real QE data, specializing in word-level tasks.

It appears that you require QE models for word-level tasks. However, we would like to kindly note that word-level QE models may be insufficient for complex tasks due to label imbalance issues.

**Comment by p1k0pan** [comment]
Thank you for releasing the Models. Could you provide a simple example to show how to use these models?

**Comment by hy5468** [comment]
Hi, while the evaluate procedure can be complex, its core steps are outlined as follows:

1. Perform inference with the model to obtain original token-level OK probabilities and sentence-level scores.

```
SRC=en
MT=de
valid_path=PREFIX FOR THE VALIDATION PATH
NJUQE=PATH TO THE NJUQE CODE
save_dir=PATH FOR SAVING OUTPUT FILES (including checkpoints to be inferred)

python $NJUQE/scripts/eval_mqm.py \
	${valid_path["${SRC}-${MT}"]} \
	--user-dir $NJUQE/njuqe \
	--task qe --criterion qe_base \
	--optimizer adam --adam-betas '(0.9,0.98)' \
	--src $SRC --mt $MT --mt-tag tag --score mqm_score --bounds mt --max-sentences 20 --dataset-impl mmap \
	--fine-tune --prepend-bos --append-eos --mask-symbol --predict-target --predict-score direct --qe-meter \
	--qe-destdir $save_dir --eval-split test --skip-invalid-size-inputs-valid-test --path $save_dir/checkpoint_best.pt --no-shuffle >>$save_dir/train.log 2>&1
```
2. Following COMET, the final sentence-level score is obtained by using $NJUQE/scripts/evaluate_word_2_sent.py to determine the optimal threshold on the validation data, then applying this best validation threshold to the test data.
```
    # find the best threshold sets on valid set
    best_thresholds, best_spearman_scores, best_scores = find_best_thresholds(
         probs, ref_scores)
    
    # applying this best validation threshold to the test data
    best_thresholds = {'critical': , 'major': , 'minor': }
```

3. The final MQM tag is obtained by using $NJUQE/scripts/submission_format_span.py to determine the optimal threshold on the validation data, then applying this best validation threshold to the test data. Alternatively, you may directly use the threshold released in the DCSQE paper.

4. For simplicity, you may directly use the original token-level OK probabilities and sentence-level scores.

**Comment by p1k0pan** [comment]
Thanks for the examples. 
I have a question about the scope of language pairs. I saw the models have suffix like zh-en or en-de. Does that mean model only trained by the specific language pairs and it is not reliable if using these model on other language pair like en-zh or en-fr?

**Comment by hy5468** [comment]
Yes, this means the model was only trained on specific language pairs, making it unreliable when applied to other language pairs. Even multilingual models like COMET are trained on several language pairs. If you want to perform Quality Estimation (QE) for other languages, I recommend generating pseudo data (such as with DCSQE) and training your own model.
